From 46f21c010979937f8deed2c5edba548bb4bd5372 Mon Sep 17 00:00:00 2001
From: t <tristan75015@gmail.com>
Date: Tue, 6 Oct 2026 06:03:41 +0200
Subject: [PATCH] generic-parallel: TDLS tests for all the configurations

The behaviours using the TDLS linear system solver are now also tested
with the cuda-clang, hip-clang, stlpar-parunseq-clang,
stlpar-parunseq-icpx and sycl-default-icpx configurations.
---
 .../generic-parallel/cuda-clang/tdls/test.check       | 11 +++++++++++
 .../hip-clang-nvidia-platform/tdls/test.check         | 11 +++++++++++
 .../stlpar-parunseq-clang/tdls/test.check             | 11 +++++++++++
 .../stlpar-parunseq-icpx/tdls/test.check              | 11 +++++++++++
 .../sycl-default-icpx/tdls/test.check                 | 11 +++++++++++
 5 files changed, 55 insertions(+)
 create mode 100644 test-suite/behaviours/generic-parallel/cuda-clang/tdls/test.check
 create mode 100644 test-suite/behaviours/generic-parallel/hip-clang-nvidia-platform/tdls/test.check
 create mode 100644 test-suite/behaviours/generic-parallel/stlpar-parunseq-clang/tdls/test.check
 create mode 100644 test-suite/behaviours/generic-parallel/stlpar-parunseq-icpx/tdls/test.check
 create mode 100644 test-suite/behaviours/generic-parallel/sycl-default-icpx/tdls/test.check

diff --git a/test-suite/behaviours/generic-parallel/cuda-clang/tdls/test.check b/test-suite/behaviours/generic-parallel/cuda-clang/tdls/test.check
new file mode 100644
index 000000000..2dc05c55d
--- /dev/null
+++ b/test-suite/behaviours/generic-parallel/cuda-clang/tdls/test.check
@@ -0,0 +1,11 @@
+@Requires {"mfront::behaviour::generic-parallel_interface::supported_configuration::cuda-clang",
+            "mfront::linear_system_solver::tdls"};
+
+@Environment {"LD_LIBRARY_PATH": "src"};
+// behaviours using the TDLS linear system solver
+@Command "@mfront@ --obuild --search-path=../.. --interface=generic --interface=generic-parallel --configuration-file=config-cuda-clang.json ../../ImplicitNortonTDLS.mfront";
+@Command "@TFEL_CLANGXX_COMPILER@ -O2 -std=c++@TFEL_CXX_STANDARD@ -x cuda --cuda-gpu-arch=native --cuda-path=@TFEL_CLANG_CUDA_PATH@ -Wno-unknown-cuda-version -I @TFEL_INCLUDE_DIR@ ../../compare.cxx -o compare -L@TFEL_CLANG_CUDA_LIBRARY_DIR@ -lcudart -ldl";
+@Command "sh ../../compare-tdls.sh ./compare src/libGenericParallelBehaviour-cuda.so";
+
+@CleanFiles {"compare"};
+@CleanDirectories {"include", "src"};
diff --git a/test-suite/behaviours/generic-parallel/hip-clang-nvidia-platform/tdls/test.check b/test-suite/behaviours/generic-parallel/hip-clang-nvidia-platform/tdls/test.check
new file mode 100644
index 000000000..c0f4f57dd
--- /dev/null
+++ b/test-suite/behaviours/generic-parallel/hip-clang-nvidia-platform/tdls/test.check
@@ -0,0 +1,11 @@
+@Requires {"mfront::behaviour::generic-parallel_interface::supported_configuration::hip-clang-nvidia-platform",
+            "mfront::linear_system_solver::tdls"};
+
+@Environment {"LD_LIBRARY_PATH": "src"};
+// behaviours using the TDLS linear system solver
+@Command "@mfront@ --obuild --search-path=../.. --interface=generic --interface=generic-parallel --configuration-file=config-hip-clang-nvidia-platform.json ../../ImplicitNortonTDLS.mfront";
+@Command "@TFEL_CLANGXX_COMPILER@ -O2 -std=c++@TFEL_CXX_STANDARD@ -x cuda --cuda-gpu-arch=native --cuda-path=@TFEL_CLANG_CUDA_PATH@ -Wno-unknown-cuda-version -I@TFEL_HIP_INCLUDE_DIR@ -D__HIP_PLATFORM_NVIDIA__ -DMFRONT_GPB_COMPARE_USE_HIP -I @TFEL_INCLUDE_DIR@ ../../compare.cxx -o compare -L@TFEL_CLANG_CUDA_LIBRARY_DIR@ -lcudart -ldl";
+@Command "sh ../../compare-tdls.sh ./compare src/libGenericParallelBehaviour-hip.so";
+
+@CleanFiles {"compare"};
+@CleanDirectories {"include", "src"};
diff --git a/test-suite/behaviours/generic-parallel/stlpar-parunseq-clang/tdls/test.check b/test-suite/behaviours/generic-parallel/stlpar-parunseq-clang/tdls/test.check
new file mode 100644
index 000000000..42b7cb616
--- /dev/null
+++ b/test-suite/behaviours/generic-parallel/stlpar-parunseq-clang/tdls/test.check
@@ -0,0 +1,11 @@
+@Requires {"mfront::behaviour::generic-parallel_interface::supported_configuration::stlpar-parunseq-clang",
+            "mfront::linear_system_solver::tdls"};
+
+@Environment {"LD_LIBRARY_PATH": "src"};
+// behaviours using the TDLS linear system solver
+@Command "@mfront@ --obuild --search-path=../.. --interface=generic --interface=generic-parallel --configuration-file=config-stlpar-parunseq-clang.json ../../ImplicitNortonTDLS.mfront";
+@Command "@TFEL_CLANGXX_COMPILER@ -O2 -march=native --std=c++@TFEL_CXX_STANDARD@ -I @TFEL_INCLUDE_DIR@ ../../compare.cxx -o compare -ldl";
+@Command "sh ../../compare-tdls.sh ./compare src/libGenericParallelBehaviour-stlpar.so";
+
+@CleanFiles {"compare"};
+@CleanDirectories {"include", "src"};
diff --git a/test-suite/behaviours/generic-parallel/stlpar-parunseq-icpx/tdls/test.check b/test-suite/behaviours/generic-parallel/stlpar-parunseq-icpx/tdls/test.check
new file mode 100644
index 000000000..5edc81fd1
--- /dev/null
+++ b/test-suite/behaviours/generic-parallel/stlpar-parunseq-icpx/tdls/test.check
@@ -0,0 +1,11 @@
+@Requires {"mfront::behaviour::generic-parallel_interface::supported_configuration::stlpar-parunseq-icpx",
+            "mfront::linear_system_solver::tdls"};
+
+@Environment {"LD_LIBRARY_PATH": "src"};
+// behaviours using the TDLS linear system solver
+@Command "@mfront@ --obuild --search-path=../.. --interface=generic --interface=generic-parallel --configuration-file=config-stlpar-parunseq-icpx.json ../../ImplicitNortonTDLS.mfront";
+@Command "@TFEL_ICPX_COMPILER@ -O2 -march=native --std=c++@TFEL_CXX_STANDARD@ -I @TFEL_INCLUDE_DIR@ ../../compare.cxx -o compare -ldl";
+@Command "sh ../../compare-tdls.sh ./compare src/libGenericParallelBehaviour-stlpar.so";
+
+@CleanFiles {"compare"};
+@CleanDirectories {"include", "src"};
diff --git a/test-suite/behaviours/generic-parallel/sycl-default-icpx/tdls/test.check b/test-suite/behaviours/generic-parallel/sycl-default-icpx/tdls/test.check
new file mode 100644
index 000000000..3102028e6
--- /dev/null
+++ b/test-suite/behaviours/generic-parallel/sycl-default-icpx/tdls/test.check
@@ -0,0 +1,11 @@
+@Requires {"mfront::behaviour::generic-parallel_interface::supported_configuration::sycl-default-icpx",
+            "mfront::linear_system_solver::tdls"};
+
+@Environment {"LD_LIBRARY_PATH": "src"};
+// behaviours using the TDLS linear system solver
+@Command "@mfront@ --obuild --search-path=../.. --interface=generic --interface=generic-parallel --configuration-file=config-sycl-default-icpx.json ../../ImplicitNortonTDLS.mfront";
+@Command "@TFEL_ICPX_COMPILER@ -fsycl -O2 -march=native --std=c++@TFEL_CXX_STANDARD@ -DMFRONT_GPB_COMPARE_USE_SYCL -I @TFEL_INCLUDE_DIR@ ../../compare.cxx -o compare -ldl";
+@Command "sh ../../compare-tdls.sh ./compare src/libGenericParallelBehaviour-sycl.so";
+
+@CleanFiles {"compare"};
+@CleanDirectories {"include", "src"};
-- 
2.43.0

